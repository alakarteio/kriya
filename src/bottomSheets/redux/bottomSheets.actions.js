export const OPEN_BOTTOMSHEETS = 'OPEN_BOTTOMSHEETS'
export const open = (code, contents) => ({ type: OPEN_BOTTOMSHEETS, payload: { code, contents } })

export const CLOSE_BOTTOMSHEETS = 'CLOSE_BOTTOMSHEETS'
export const close = code => ({ type: CLOSE_BOTTOMSHEETS, payload: code })

export const TOGGLE_BOTTOMSHEETS = 'TOGGLE_BOTTOMSHEETS'
export const toggle = code => ({ type: TOGGLE_BOTTOMSHEETS, payload: code })
