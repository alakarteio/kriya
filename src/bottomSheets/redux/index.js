import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
