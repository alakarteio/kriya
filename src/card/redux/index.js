export { default } from './card'
export * from './card.actions'
export * from './card.selectors'
