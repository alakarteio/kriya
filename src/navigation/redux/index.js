export { default } from './navigation'
export * from './navigation.actions'
export * from './navigation.selectors'
