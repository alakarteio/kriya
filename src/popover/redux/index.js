export { default } from './popover'
export * from './popover.actions'
export * from './popover.selectors'
