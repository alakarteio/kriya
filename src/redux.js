export { default as cards } from './card/redux'
export { default as navigation } from './navigation/redux'
export { default as popover } from './popover/redux'
export { default as bottomSheets } from './bottomSheets/redux'
export { default as tabs } from './tabs/redux'
