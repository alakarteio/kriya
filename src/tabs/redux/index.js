export { default } from './tabs'
export * from './tabs.actions'
export * from './tabs.selectors'
