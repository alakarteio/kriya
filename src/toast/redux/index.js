import toast from './toast'
import * as actions from './toast.actions'
import * as selectors from './toast.selectors'

Object.assign(toast, actions, selectors)

export * from './toast.actions'
export * from './toast.selectors'
export default toast

