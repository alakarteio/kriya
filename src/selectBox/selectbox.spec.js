/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import mockComponent from '__mocks__/component'
import SelectBox from './selectBox.jsx'
import SelectBoxContainer from './selectBox.container'

jest.mock('kriya-select', () => mockComponent('kriya-select'))

const Decorated = reduxForm({ form: 'testForm' })(SelectBox)
const DecoratedContainer = reduxForm({ form: 'formCont' })(SelectBoxContainer)

const snap = ({ name = 'name', ...rest }) => {
  const store = createStore(() => ({}))
  const component = renderer.create(
    <Provider store={store}>
      <Decorated {...rest} name={name} />
    </Provider>,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/SelectBox', () => {
  describe('graphical (JSX)', () => {
    it('should have a default behaviour', () => snap({}))
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should add options', () => snap(
      {
        options: [
          { label: 'Chirurgiens cardio-vasculaires et thoraciques', value: '1' },
          { label: 'Chirurgiens de la face et du cou', value: '2' },
          { label: 'Chirurgiens de la main', value: '3' },
        ],
      }),
    )

    describe('labels', () => {
      it('should print a label', () => snap({ label: 'my label' }))
      it('should print a * on label when input is required', () => snap({ required: true, label: 'my label' }))
    })
  })

  describe('container', () => {
    const snapContainer = (state, props) => {
      const store = createStore(() => (
        {
          form: {
            form1: {
              values: {
                field1: '1',
              },
            },
          },
        }
      ))
      const component = renderer.create(
        <Provider store={store}>
          <DecoratedContainer {...props} />
        </Provider>,
      )

      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    }

    it('should pass value of redux form', () => snapContainer({}, {
      form: 'form1',
      name: 'field1',
      options: [
        { label: 'Chirurgiens cardio-vasculaires et thoraciques', value: '1' },
        { label: 'Chirurgiens de la face et du cou', value: '2' },
        { label: 'Chirurgiens de la main', value: '3' },
      ],
    }))
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
