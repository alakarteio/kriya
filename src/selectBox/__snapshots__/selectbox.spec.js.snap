// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/SelectBox container should pass value of redux form 1`] = `
<div
  className="select"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":true,"form":"form1","initialized":false,"invalid":false,"pristine":false,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"options":[{"label":"Chirurgiens cardio-vasculaires et thoraciques","value":"1"},{"label":"Chirurgiens de la face et du cou","value":"2"},{"label":"Chirurgiens de la main","value":"3"}],"pure":true,"array":{},"value":"1"}}
  </div>
</div>
`;

exports[`common/SelectBox graphical (JSX) labels should print a * on label when input is required 1`] = `
<div
  className="select"
  style={Object {}}
>
  <label
    htmlFor="name"
  >
    my label
    *
  </label>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{},"options":[]}}
  </div>
</div>
`;

exports[`common/SelectBox graphical (JSX) labels should print a label 1`] = `
<div
  className="select"
  style={Object {}}
>
  <label
    htmlFor="name"
  >
    my label
  </label>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{},"options":[]}}
  </div>
</div>
`;

exports[`common/SelectBox graphical (JSX) should add custom className 1`] = `
<div
  className="select custom"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{},"options":[]}}
  </div>
</div>
`;

exports[`common/SelectBox graphical (JSX) should add custom style 1`] = `
<div
  className="select"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{},"options":[]}}
  </div>
</div>
`;

exports[`common/SelectBox graphical (JSX) should add options 1`] = `
<div
  className="select"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"options":[{"label":"Chirurgiens cardio-vasculaires et thoraciques","value":"1"},{"label":"Chirurgiens de la face et du cou","value":"2"},{"label":"Chirurgiens de la main","value":"3"}],"pure":true,"array":{}}}
  </div>
</div>
`;

exports[`common/SelectBox graphical (JSX) should have a default behaviour 1`] = `
<div
  className="select"
  style={Object {}}
>
  <div
    className="kriya-select"
  >
    {"name":"kriya-select","props":{"simpleValue":true,"anyTouched":false,"asyncValidating":false,"dirty":false,"form":"testForm","initialized":false,"invalid":false,"pristine":true,"submitting":false,"submitFailed":false,"submitSucceeded":false,"valid":true,"pure":true,"array":{},"options":[]}}
  </div>
</div>
`;
